clients/src/main/java/org/apache/kafka/clients/producer/Partitioner.java

```diff
@@ -26,6 +26,15 @@
  */
 public interface Partitioner extends Configurable, Closeable {
 
+    /**
+     * Indicate if the given topic is handled.  Returning {@code false} will cause the Producer to fallback to default partitioning.
+     *
+     * @param topic The topic name
+     */
+    default boolean partitionsTopic(String topic) {
+        return true;
```

Comment:
why is it true by default?

Reply:
I think the idea is that this would be implemented differently by different custom partitioners. As default though there is no change from current behavior.